src/Test/QuickCheck/StateMachineModel.hs

```diff
@@ -38,23 +40,25 @@ import           Test.QuickCheck.StateMachineModel.Utils
 
 ------------------------------------------------------------------------
 
+
+
 liftSem
   :: forall cmd resp ix ref m
   .  (Monad m, Enum ix, Typeable ref, RefKit cmd)
   => Typeable resp
   => Show resp
   => (cmd resp ref -> m resp)
-  -> cmd resp ix -> StateT [ref] m resp
+  -> cmd resp ix -> StateT (Map Int ref) m resp
```

Comment:
Why is this `Int` rather than `ix`? because you need to compute the next var? Why not take in a function:
`new :: Map ix ref -> ix`, and use that instead of `M.size`?